### DISCUSSION TITLE
Pump Status/Info in Web Service

### DISCUSSION THREAD

**Discussion by ranvens** [NightscoutFoundation/xDrip#3816] [Q&A]
Is it possible to get the "Pump Status" info via the web service of xdrip (either json or /pebble)?

I'm running android x86 in a VM on Proxmox, where I have xdrip installed and connected to my child's Carelink follower account. 
I then use the web service of xdrip to obtain information to tie into my Home Assistant and N8N instances which can set of a number of automations depending on the criteria (EG: Alert on increasing bgl over Alexa / Speakers, flash lights, alarms on low etc). Overall this has helped us better manage our t1d'er. 

The pump status info would be handy to be able to use for additional automations / reminders. We have recently migrated to the Medtronic 7 day sets, however - they still have not updated pump / carelink software to alert for the 7 day set reminder (still max of 3 days). I could use the total insulin info of the pump to create an automation where if it goes above X units of insulin reset timer and create an alert for 7 days time etc. It would also be beneficial to be able to report on the pump battery level to prewarn when its below a certain value.

**Comment by Navid200** [comment]
As far as I know xDrip does not communicate with any pump.
Have you looked the AAPS project?

**Reply by ranvens** [reply]
It definitely does :) 
![Screenshot 2024-12-16 183425](https://github.com/user-attachments/assets/c24d06a6-f6b2-4743-bc48-2d3c009b5abb)

I'll check out the AAPS project when i have some spare time :)
